presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -1517,6 +1517,32 @@ public void testGroupingCombinationsDistinct()
     }
 
     @Test
+    public void testRollupOverUnion()
+            throws Exception
+    {
+        MaterializedResult actual = computeActual("" +
```

Comment:
I think you can write this as a query and avoid manually creating the result set:
```java
assertQuery(
...,
"VALUES ('P', 21470000), (NULL, 21470000), ...")